### PR TITLE
Testfreeze

### PR DESCRIPTION
**Pull request by YingClara92**

### CODE REVIEW

File: Models/AutoEncoder3D.py [hunk 1]
```diff
@@ -1,87 +1,167 @@
...
+        plt.show()
+        return prediction
+
     def configure_optimizers(self):
         optimizer = torch.optim.Adam(self.parameters(), lr=1e-3)
         scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=5, gamma=0.5)
         return [optimizer], [scheduler]
 
+
 if __name__ == "__main__":
```
Comment (cacof1) [inline Models/AutoEncoder3D.py:87]: "main" are great testing but please don't leave them here when you are satisfied

File: Models/AutoEncoder3D.py [hunk 2]
```diff
@@ -1,87 +1,167 @@
...
+
+        summary(self.model.to('cuda'), (config['MODEL']['batch_size'], 1, *config['DATA']['dim']))
         self.accuracy = torchmetrics.AUC(reorder=True)
-        self.loss_fcn = torch.nn.BCEWithLogitsLoss()
+        self.loss_fcn = torch.nn.MSELoss()
 
-    def forward(self, x):
-        return self.model(x)
+    def forward(self, datadict):
+        key = list(datadict.keys())[0]
```
Comment (cacof1) [inline Models/AutoEncoder3D.py:44]: this is not great, it ensure that you always use the first model_dict "[0]", but what if you want to do an autoencoder on Dose? Or what if someone define modality as Dose, CT, etc but wants to do it on CT?

File: Models/Classifier3D.py [hunk 3]
```diff
@@ -27,8 +27,8 @@ def __init__(self, config, module_str):
             torch.nn.Flatten(),
         )
         self.model.apply(self.weights_init)
-        # summary(self.model.to('cuda'), (config['MODEL']['batch_size'], 1, *config['DATA']['dim']),
-        #         col_names=["input_size", "output_size"], depth=5)
+        summary(self.model.to('cuda'), (config['MODEL']['batch_size'], 1, *config['DATA']['dim']),
```
Comment (cacof1) [inline Models/Classifier3D.py:30]: you can kick out the summary now that we are using MONAI

**Review by cacof1 — COMMENTED** [review]
Few comments to address still.

File: Models/AutoEncoder3D.py (hunk 2 above)
Comment (YingClara92) [reply Models/AutoEncoder3D.py:44]: For the autoencoder, I want to use one modality each time, CT or Dose.  so the setting files just have one key.

File: Models/AutoEncoder3D.py (hunk 1 above)
Comment (YingClara92) [reply Models/AutoEncoder3D.py:87]: I am not sure if use autoencoder will help, when I am sure, I will make another file

File: Models/AutoEncoder3D.py [hunk 4]
```diff
@@ -59,12 +62,14 @@ def validation_step(self, batch, batch_idx):
         loss = self.loss_fcn(prediction, image_dict[key])
 
         if batch_idx < 3:
-            im_st = image_dict[key][0, 0, 5, :, :]
-            plt.imshow(im_st.cpu())
+            im_st = image_dict[key][0, :, :, :, :]
+            self.logger.log_image(im_st, 'standard image', self.current_epoch)
+            plt.imshow(im_st[0, 5, :, :].cpu(), cmap='gray')
```
Comment (cacof1) [inline Models/AutoEncoder3D.py:67]: Please don't leave comment like this in the code (imshow). The code online should not be for testing

File: Models/AutoEncoder3D.py (hunk 1 above)
Comment (cacof1) [reply Models/AutoEncoder3D.py:93]: Still the code online should not be for testing phase. Please remove the if__main__ from it and keep it in a local file.

File: Models/AutoEncoder3D.py (hunk 2 above)
Comment (cacof1) [reply Models/AutoEncoder3D.py:47]: You may want to only use one modality, but some people might want to use the autoencoder weights loaded, so will need the autoencoder model. Just in general, it's bad practice to have hard coded variables.

**Review by cacof1 — APPROVED** [review]

**Pull request merged**